test_apps/smokeTests/testGrid.cpp

```diff
@@ -0,0 +1,274 @@
+    double t0 = Wasp::GetTime();
+
+    OptionParser op;
+    InitializeOptions( argc, argv, op );
+
+    std::cout << std::fixed << std::showpoint;
+    std::cout << std::setprecision(4);
+
+    std::vector< double > minu = { 
+        opt.extents[X], 
+        opt.extents[Y], 
+        opt.extents[Z] 
+    };
+    std::vector< double > maxu = { 
+        opt.extents[X+3], 
+        opt.extents[Y+3], 
+        opt.extents[Z+3] 
+    };
+
+    std::vector< size_t > bs2d = { opt.bs[X], opt.bs[Y] }; 
+    std::vector< double > minu2d = { minu[X], minu[Y] };
+    std::vector< double > maxu2d = { maxu[X], maxu[Y] };
+
+    std::vector< size_t > dims2d = { opt.dims[X], opt.dims[Y] };
+
+    int rc            = 0;
+    int regularRC     = 0;
+    int stretchedRC   = 0;
+    int layeredRC     = 0;
+    int curvilinearRC = 0;
+
+    // Test Regular Grid
+    if ( std::find( opt.grids.begin(), opt.grids.end(), "Regular" ) != opt.grids.end() ) {
+        std::vector< float* > rgBlks  = AllocateBlocks( opt.bs, opt.dims );
+        RegularGrid* regularGrid = new RegularGrid( opt.dims, opt.bs, rgBlks, minu, maxu );
+        regularRC = RunTests( regularGrid, opt.arrangements, opt.minValue, opt.maxValue );
+        delete regularGrid;
+    }
+
+    // Test Stretched Grid
+    if ( std::find( opt.grids.begin(), opt.grids.end(), "Stretched" ) != opt.grids.end() ) {
+        StretchedGrid* stretchedGrid = MakeStretchedGrid( opt.dims, opt.bs, minu, maxu );
+        stretchedRC = RunTests( stretchedGrid, opt.arrangements, opt.minValue, opt.maxValue );
+        delete stretchedGrid;
+    }
+
+    // Test Layered Grid
+    if ( std::find( opt.grids.begin(), opt.grids.end(), "Layered" ) != opt.grids.end() ) {
+        LayeredGrid* layeredGrid = MakeLayeredGrid( opt.dims, opt.bs, minu, maxu );
+        layeredRC = RunTests( layeredGrid, opt.arrangements, opt.minValue, opt.maxValue );
+        delete layeredGrid;
+    }
+   
+    // Test Curvilinear Grid
+    if ( std::find( opt.grids.begin(), opt.grids.end(), "Curvilinear" ) != opt.grids.end() ) {
+        CurvilinearGrid* curvilinearGrid;
+        curvilinearGrid = MakeCurvilinearTerrainGrid( opt.bs, minu, maxu, opt.dims);
+        curvilinearRC = RunTests( curvilinearGrid, opt.arrangements, opt.minValue, opt.maxValue );
+        delete curvilinearGrid;
+    }
+
+    if ( regularRC != 0 ) {
+        cout << "Errors occurred while testing Grid::RegularGrid." << endl;
+        rc = -1;
+    }
+    if ( stretchedRC != 0 ) {
+        cout << "Errors occurred while testing Grid::StretchedGrid." << endl;
+        rc = -1;
+    }
+    if ( layeredRC != 0 ) {
+        cout << "Errors occurred while testing Grid::LayeredGrid." << endl;
+        rc = -1;
+    }
+    if ( curvilinearRC != 0 ) {
+        cout << "Errors occurred while testing Grid::CurvilinearGrid." << endl;
+        rc = -1;
+    }
+
+    double t1 = Wasp::GetTime();
+    cout << "Elapsed time: " << t1-t0 << endl;
+
+    DeleteHeap();
+
+    return rc;
```

Comment:
The convention for executables is to return 0 for success and 1 for failure. For whatever reasons system calls return a negative int on failure, but executables return a value between 0 and 255, with 0 indicating success and 1 indicating failure. The remaining 254 values are reserved for other forms of status.

Reply:
Thanks - fixed.